controllers/batchtransformjob/batchtransformjob_controller.go

```diff
@@ -316,6 +313,11 @@ func (r *BatchTransformJobReconciler) handleSageMakerApiFailure(ctx reconcileReq
 			ctx.Log.Info("SageMaker rate limit exceeded, will retry", "err", awsErr)
```

Comment:
Are there other cases where we need to requeue ?

Reply:
Yes, there are some cases where we should requeue but we do not. Any 4xx on [this page](https://docs.aws.amazon.com/sagemaker/latest/dg/CommonErrors.html) should be retried. I don't want to go about refactoring this without tests as it is a significant change. The comment on [this task](https://issues.amazon.com/issues/P31245685) explains that non-429 4xx errors are actually treated as 404s, which is a bug.

I would love to start working on that task this sprint, let's discuss in planning.